src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleGroupSyncMonitor.scala

```diff
@@ -132,28 +138,45 @@ class GoogleGroupSyncMonitorActor(
 
   override def receive = {
     case StartMonitorPass =>
-      logger.debug(s"Pulling messages off: $pubSubSubscriptionName")
+      logger.info(s"Pulling messages off: $pubSubSubscriptionName")
       // start the process by pulling a message and sending it back to self
       pubSubDao.pullMessages(pubSubSubscriptionName, 1).map(_.headOption) pipeTo self
 
     case Some(message: PubSubMessage) =>
-      logger.debug(s"received sync message: $message")
-      import Tracing._
-      trace("GoogleGroupSyncMonitor-PubSubMessage") { span =>
+      logger.info(s"received sync message: $message")
+
+      traceIO("GoogleGroupSyncMonitor-PubSubMessage", SamRequestContext()) { samRequestContext =>
+        // every 15s reset the deadline to 60s in the future, canceled when IO is done
+        // 15s so we get a few tries at extending before deadline occurs
+        val deadlineExtender = system.scheduler.scheduleAtFixedRate(Duration.Zero, 15 seconds, self, ExtendDeadline(message.ackId, 60, samRequestContext))
+
         val groupId: WorkbenchGroupIdentity = parseMessage(message)
+        samRequestContext.parentSpan.foreach(
+          _.putAttributes(
+            Map("groupId" -> AttributeValue.stringAttributeValue(groupId.toString), "messageId" -> AttributeValue.stringAttributeValue(message.ackId)).asJava
+          )
+        )
         groupSynchronizer
           .synchronizeGroupMembers(
             groupId,
-            samRequestContext = SamRequestContext(Option(span))
-          ) // Since this is an internal pub/sub call, we have to start a new SamRequestContext.
-          .toTry
-          .map(sr => sr.fold(t => FailToSynchronize(t, message.ackId), x => ReportMessage(x, message.ackId))) pipeTo self
-      }
+            samRequestContext = samRequestContext
+          )
+          .redeem(t => FailToSynchronize(t, message.ackId), x => ReportMessage(x, message.ackId))
+          .guarantee(IO(deadlineExtender.cancel()))
```

Comment:
Some scala IO explanation for other reviewers:
`redeem` takes two functions, one to execute in the failure case and one to execute in the success case. 
`guarantee` will always run, no matter if the IO is successful, failed, or cancelled. This way, the `deadlineExtender` is always cancelled once the IO is finished.